indexer/src/configs.rs

```diff
@@ -0,0 +1,168 @@
+    propagate_version(true),
+    next_line_help(true)
+)]
+pub(crate) struct Opts {
+    /// Connection string to connect to the PostgreSQL Database to fetch AlertRules from
+    #[clap(long, env)]
+    pub database_url: String,
+    /// Enabled Indexer for Explorer debug level of logs
+    #[clap(long)]
+    pub debug: bool,
+    /// Switches indexer to non-strict mode (skips Receipts without parent Transaction hash, stops storing AccountChanges and AccessKeys)
+    #[clap(long)]
+    pub non_strict_mode: bool,
+    /// Sets the concurrency for indexing. Note: concurrency (set to 2+) may lead to warnings due to tight constraints between transactions and receipts (those will get resolved eventually, but unless it is the second pass of indexing, concurrency won't help at the moment).
+    #[clap(long, default_value = "1")]
+    pub concurrency: std::num::NonZeroU16,
+    /// Port to enable metrics/health service
+    #[clap(long, short, env, default_value_t = 3030)]
+    pub port: u16,
+    /// Chain ID: testnet or mainnet
+    #[clap(subcommand)]
+    pub chain_id: ChainId,
+}
+
+#[derive(Subcommand, Debug, Clone)]
+pub enum ChainId {
+    #[clap(subcommand)]
+    Mainnet(StartOptions),
+    #[clap(subcommand)]
+    Testnet(StartOptions),
+}
+
+#[allow(clippy::enum_variant_names)]
+#[derive(Subcommand, Debug, Clone)]
+pub enum StartOptions {
+    /// Start from specific block height
+    FromBlock { height: u64 },
+    /// Start from interruption (last_indexed_block value from Redis)
+    FromInterruption,
+    /// Start from the final block on the network (queries JSON RPC for finality: final)
+    FromLatest,
+}
+
+impl Opts {
+    /// Returns [StartOptions] for current [Opts]
+    pub fn start_options(&self) -> &StartOptions {
+        match &self.chain_id {
+            ChainId::Mainnet(start_options) | ChainId::Testnet(start_options) => start_options,
+        }
+    }
+
+    pub fn rpc_url(&self) -> &str {
+        match self.chain_id {
+            ChainId::Mainnet(_) => "https://rpc.mainnet.near.org",
+            ChainId::Testnet(_) => "https://rpc.testnet.near.org",
+        }
+    }
+}
+
+impl Opts {
+    pub async fn to_lake_config(&self) -> near_lake_framework::LakeConfig {
+        let config_builder = near_lake_framework::LakeConfigBuilder::default();
+
+        match &self.chain_id {
+            ChainId::Mainnet(_) => config_builder
+                .mainnet()
+                .start_block_height(get_start_block_height(self).await),
+            ChainId::Testnet(_) => config_builder
+                .testnet()
+                .start_block_height(get_start_block_height(self).await),
+        }
+        .build()
+        .expect("Failed to build LakeConfig")
+    }
+}
+
+async fn get_start_block_height(opts: &Opts) -> u64 {
+    match opts.start_options() {
+        StartOptions::FromBlock { height } => *height,
+        StartOptions::FromInterruption => {
+            let pool = models::establish_connection(&opts.database_url);
```

Comment:
We create the pool here and drop it at the end of the function.
Shouldn't we create it outside, in main?
I think it's better to pass to this function only `pool` and `start_options`

Reply:
This function is independent of the `main`. I did this intentionally, but now I see we can improve it, though it's out of the scope of this PR.

Reply:
Anyway, when we call it, we need to have `pool` ready